### PR TITLE
REP 3175 Add Header Filter config update

### PR DESCRIPTION
**Pull request by Mario-Lopez**
Previously the Add Header filter required you to configure headers for both the request and the response.  I updated the XSD to allow a user to configure just one of them if they wanted to.  The code already handles having just one of them configured.

There's an existing functional test with both configured.  I added two additional tests with just the request and just the response to ensure it worked.

### CODE REVIEW

File: repose-aggregator/components/filters/add-header/src/main/resources/META-INF/schema/config/add-header.xsd [hunk 1]
```diff
@@ -30,8 +30,8 @@
 
     <xs:complexType name="AddHeadersConfig">
         <xs:all>
-            <xs:element name="request" type="HttpMessage"/>
-            <xs:element name="response" type="HttpMessage"/>
+            <xs:element name="request" type="HttpMessage" minOccurs="0"/>
+            <xs:element name="response" type="HttpMessage" minOccurs="0"/>
```
Comment (wdschei) [inline repose-aggregator/components/filters/add-header/src/main/resources/META-INF/schema/config/add-header.xsd]: This should probably have the logic to ensure that there is at least one of them.

Comment (Mario-Lopez) [inline repose-aggregator/components/filters/add-header/src/main/resources/META-INF/schema/config/add-header.xsd]: I totally agree.  Do you know of an easy way to do that?  I only found two unideal ways of doing it on the web.  The less insane option was to do an assertion.  I vaguely remember seeing some other filter's doc mention having such a case of not configuring anything and the filter just being a "pass through", so I figured there was precedent.  Thoughts?

File: repose-aggregator/components/filters/add-header/src/main/resources/META-INF/schema/config/add-header.xsd [hunk 2]
```diff
@@ -30,8 +30,8 @@
...
         </xs:all>
```
Comment (wdschei) [inline repose-aggregator/components/filters/add-header/src/main/resources/META-INF/schema/config/add-header.xsd]: Something like:
```
<xs:assert vc:minVersion="1.1"
           test="count(*)>=1"
           xerces:message="At least one header must be defined."
           saxon:message="At least one header must be defined."/>
```

Comment (Mario-Lopez) [inline repose-aggregator/components/filters/add-header/src/main/resources/META-INF/schema/config/add-header.xsd]: Sounds good.  I'll add that in.  Thanks!

File: repose-aggregator/components/filters/add-header/src/main/resources/META-INF/schema/config/add-header.xsd [hunk 3]
```diff
@@ -30,9 +33,13 @@
...
+        <xs:assert vc:minVersion="1.1"
+                   test="count(*)>=1"
+                   xerces:message="At least one header must be defined."
+                   saxon:message="At least one header must be defined."/>
```
Comment (wdschei) [inline repose-aggregator/components/filters/add-header/src/main/resources/META-INF/schema/config/add-header.xsd]: Was there a test to make sure this worked?

Comment (jennyvo) [inline repose-aggregator/components/filters/add-header/src/main/resources/META-INF/schema/config/add-header.xsd]: will check and add if need

Comment (wdschei) [inline repose-aggregator/components/filters/add-header/src/main/resources/META-INF/schema/config/add-header.xsd]: It looks like this is the only thing missing before I can sign off on it.

**Pull request merged**